docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+    {
+        return a / b;
+    }
+
+    public static double Divide(double a, double b)
+    {
+        return a / b;
+    }
+}
+```
+
+The sample library above doesn't do much, it supports four major arithmetic operations `add`, `subtract`, `multiply` and `divide` on `int` and `double` datatypes.
+Now let's assume this is one huge library that is going to have a ton of developers, we'll need to add some comments to make it easier to understand.
+In this sample we'll use multiline comments for class definitions and single line comments otherwise, these are not imposed conventions however,
+so feel free to use the kind of comment the situation requires.
+
+Here's the updated library code with comments added:
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+public class Math
+{
+    // Adds two integers and returns the result
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    // Adds two doubles and returns the result
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    // Subtracts an integer from another and returns the result
+    public static int Subtract(int a, int b)
+    {
+        return a - b;
+    }
+
+    // Subtracts a double from another and returns the result
+    public static double Subtract(double a, double b)
+    {
+        return a - b;
+    }
+
+    // Multiplies two intergers and returns the result
+    public static int Multiply(int a, int b)
+    {
+        return a * b;
+    }
+
+    // Multiplies two doubles and returns the result
+    public static double Multiply(double a, double b)
+    {
+        return a * b;
+    }
+
+    // Divides an integer by another and returns the result
+    public static int Divide(int a, int b)
+    {
+        return a / b;
+    }
+
+    // Divides a double by another and returns the result
+    public static double Divide(double a, double b)
+    {
+        return a / b;
+    }
+}
+```
+
+And there you have it! now any new developer can easily get quick insight into what each method and the overall class does.
+
+Next we want to be able to create an API reference document from our code for third party developers who use our library but don't have access to the source code.
+As mentioned earlier XML documentation tags can be used to achieve this, I will now introduce you to the standard XML tags the C# compiler supports.
+
+### &lt;summary&gt;
+
+First off is the `<summary>` tag and as the name suggests it is used to add brief information about a type or member.
```

Comment:
You're using 3rd person (it is used) and first person (I will now introduce).

This will read better if it's all in second person.